ucte/ucte-converter/src/main/java/com/powsybl/ucte/converter/util/UcteValidation.java

```diff
@@ -0,0 +1,68 @@
+/**
+ * Copyright (c) 2020, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.ucte.converter.util;
+
+import com.powsybl.iidm.network.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * @author Anne Tilloy {@literal <anne.tilloy at rte-france.com>}
+ */
+public final class UcteValidation {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(UcteValidation.class);
+
+    private UcteValidation() {
+    }
+
+    public static void run(Network network) {
+        checkNodesReferenceVoltage(network);
+        checkGenerators(network);
+        checkTransformers(network);
+    }
+
+    // Data: node voltage
+    private static void checkNodesReferenceVoltage(Network network) {
+        double nominalVoltage;
+        double voltage;
+        for (Bus bus : network.getBusView().getBuses()) {
+            nominalVoltage = bus.getVoltageLevel().getNominalV();
+            voltage = bus.getV();
+            if (voltage < 0.8 * nominalVoltage || voltage > 1.2 * nominalVoltage) {
+                LOGGER.warn("Wrong value for reference voltage");
```

Comment:
This log doesn't give enough information: If plenty of my buses have a bad nominalVoltage, I won't be able to find which one. You have to add contextual information in the log such as the busId, the nominal voltage, the lower et upper bound.